utils/geoutils.py

```diff
@@ -305,3 +306,88 @@ def gdf_mean_vertices_nb(gdf: gpd.GeoDataFrame):
             logging.warning(f"Only supports MultiPolygon or Polygon. \nGot {geom.geom_type}")
     mean_ext_vert_nb = np.mean(vertices_per_polygon)
     return mean_ext_vert_nb
+
+
+def mask_nodata(img_patch: Union[str, Path], gt_patch: Union[str, Path], nodata_val: int, mask_val: int = 255) -> None:
+    """
+    Masks label raster file with "ignore_index" value where pixels are "nodata" in the corresponding raster image.
+    Args:
+        img_patch: raster tile image path
+        gt_patch: raster tile label path
+        nodata_val: nodata value
+        mask_val: masking value (255 by default)
+
+    Returns:
+        Masks label tile or None if no nadata pixels
+    """
+    image_ds = gdal.Open(str(img_patch), gdalconst.GA_ReadOnly)
```

Comment:
Ideally, GDAL could be repalced where possible with rasterio, as this is one of rasterio's main goal:
https://rasterio.readthedocs.io/en/latest/contributing.html#design-principles

According to rasterio's doc, there's also some risk in using both gdal and rasterio in our scripts: https://rasterio.readthedocs.io/en/latest/topics/switch.html#mutual-incompatibilities

Reply:
Particularly for this function, I was choosing between rasterio and gdal and found gdal implementation more concise and clear.
To mask the label patch in case of rasterio, you need to read the array, save a profile, then create a new file (which I don't want to do) and then save the data in it.
GDAL provides updatinf mechanism such as gdalconst.GA_Update, so you can just rewrite the array into the same file.